docs/misc/update-wipes.md

```diff
@@ -0,0 +1,52 @@
+# Firmware update and device wipe
+
+This document describes under which circumstances the device gets wiped during a firmware
+update.
+
+## Trezor 1
+
+The device gets **wiped**:
+- If the firmware to be installed is unsigned.
+- If the present firmware is unsigned.
+- If the firmware to be installed has lower version than the current firmware's _FIX_
+version [1].
+
+The device gets **wiped on every reboot**:
+- If the firmware's debug mode is turned on.
+
+## Trezor T
+
+In Trezor T this works a bit differently, we have introduced so-called vendors headers.
+Each firmware has its vendor header and this vendor header is signed by SatoshiLabs. The
+actual firmware is signed by the vendor header's key. That means that all firmwares are
+signed by _someone_ to be able to run on Trezor T.
+
+We currently have two vendors:
+
+1. SatoshiLabs
+2. UNSAFE DO NOT USE
+
+As the names suggest, the first one is the official SatoshiLabs vendor header and all
+public firmwares are signed with that. The second one is meant for generic audience; if
+you build firmware this vendor header is automatically applied and the firmware is signed
+with it (see `tools/headertool.py`).
+
+The device gets **wiped**:
+- If the firmware to be installed is from different vendor than the present firmware [2].
+- If the firmware to be installed has lower version than the current firmware's _FIX_
+version [1].
+
+The device gets **wiped on every reboot**:
+- If the firmware's debug mode is turned on.
+
+----
+
+[1] Short version: it depends. Long version: firmware's version is specified in the
```

Comment:
what depends?
also not sure if this is the place to talk about version.h
suggest instead something like:

> Firmware headers specify a _fix version_, which is the lowest version to which that firmware can be downgraded without wiping storage. This is typically used in case the internal storage format is changed. E.g., in version 2.1.4, Shamir backup was introduced and firmwares lower than 2.1.4 would not understand it.
> It can also be used to enforce security fixes.

(probably not a good example though)

Reply:
I just wanted to provide a brief tl;dr for people not interested in the details (mainly QA). So I wanted to say "sometimes it does, sometimes it does not". But maybe its more confusing than anything?

And I agree an example would be nice. I think wipe code as introduced in 2.2.0 is a good example?

Reply:
Done, check it out.